rest_framework_simplejwt/views.py

```diff
@@ -28,10 +36,64 @@ def post(self, request, *args, **kwargs):
         except TokenError as e:
             raise InvalidToken(e.args[0])
 
-        return Response(serializer.validated_data, status=status.HTTP_200_OK)
+        response = Response(serializer.validated_data, status=status.HTTP_200_OK)
+
+        if api_settings.AUTH_COOKIE:
+            response = self.set_cookies(response, serializer.validated_data)
+
+        return response
 
+    def set_cookies(self, response, data):
+        return response
 
-class TokenObtainPairView(TokenViewBase):
+
+class TokenRefreshViewBase(TokenViewBase):
+    def extract_token_from_cookie(self, request):
+        return request
+
+    def post(self, request, *args, **kwargs):
+        if api_settings.AUTH_COOKIE:
+            request = self.extract_token_from_cookie(request)
+        return super().post(request, *args, **kwargs)
+
+
+class TokenCookieViewMixin:
+    def extract_token_from_cookie(self, request):
+        token = request.COOKIES.get('{}_refresh'.format(api_settings.AUTH_COOKIE))
+        if not token:
+            raise NotAuthenticated(detail=_('Refresh cookie not set. Try to authenticate first.'))
+        else:
+            request.data['refresh'] = token
+        return request
+
+    def set_cookies(self, response, data):
+        expires = self.get_refresh_token_expiration()
+        response.set_cookie(
+            api_settings.AUTH_COOKIE, data['access'],
+            expires=expires,
+            domain=api_settings.AUTH_COOKIE_DOMAIN,
+            path=api_settings.AUTH_COOKIE_PATH,
+            secure=api_settings.AUTH_COOKIE_SECURE or None,
+            httponly=True,
+            samesite=api_settings.AUTH_COOKIE_SAMESITE,
+        )
+        if 'refresh' in data:
+            response.set_cookie(
+                '{}_refresh'.format(api_settings.AUTH_COOKIE), data['refresh'],
+                expires=expires,
+                domain=None,
+                path=reverse('token_refresh'),
+                secure=api_settings.AUTH_COOKIE_SECURE or None,
+                httponly=True,
+                samesite='Strict',
```

Comment:
Why do you set the samesite atribute to 'strict'? 
I have my client and my API on two distinct domains so I can't refresh my tokens.
I think you may use  api_settings.AUTH_COOKIE_SAMESIT constant to set the attribute.

Reply:
the settings api_settings.AUTH_COOKIE_SAMESIT just restric the visibility of the cookie. So when using the front and back on the same pc it wont show the refresh token becaus it's samesite strict. 

using samesite lax it will be visible when using the front and back on the same pc. 

Aside the visibility, this don't prevent the front/browser of sending the refresh token when used on different servers.

Reply:
what about the `domain=None` on the creation of `{AUTH_COOKIE}_refresh`?

Reply:
same here use domain.com for the front-end and api.domain.com for the backend. i rather have the cookie in domain.com

Reply:
Same here, I'd prefer this to not be hardcoded. Current implementation means backend `api.domain.com` and frontend `domain.com` won't work, which a lot of people use.

I would prefer this:
`domain=None` -> `domain=api_settings.AUTH_COOKIE_DOMAIN`
`samesite='Strict'` -> `domain=api_settings.AUTH_COOKIE_SAMESITE`

or at least a separate setting for the refresh token so it's configurable.